packages/sdk-js/src/GrowthBook.ts

```diff
@@ -488,12 +501,46 @@ export class GrowthBook<
 
     // Apply new changes
     if (result.inExperiment) {
-      const undo = this._applyDOMChanges(result.value);
-      if (undo) {
-        this._activeAutoExperiments.set(experiment, {
-          undo,
-          valueHash,
-        });
+      if (result.value.urlRedirect) {
+        const currUrl = new URL(this._getContextUrl());
+        const redirectUrl = new URL(result.value.urlRedirect);
+        mergeUrlSearchParams(currUrl.searchParams, redirectUrl.searchParams);
+
+        const url = experiment.persistQueryString
+          ? redirectUrl.toString()
+          : result.value.urlRedirect;
+        if (
+          experiment.urlPatterns &&
+          isURLTargeted(url, experiment.urlPatterns)
+        ) {
```

Comment:
This makes me thing - what if the experiment doesn't have `urlPatterns` at all?  That could be pretty bad.

Reply:
Hmm how do we handle that possibility with a visual changesets currently?